Firestore/core/src/firebase/firestore/remote/grpc_root_certificate_finder_apple.mm

```diff
@@ -37,6 +37,18 @@
 
 namespace {
 
+std::string DebugBundleIdentifier(NSBundle* bundle) {
+  if (!bundle) return "(nil bundle)";
+
+  NSString* identifier = [bundle bundleIdentifier];
+  if (!identifier) {
```

Comment:
Since this is still Objective-C, this function could be:

```
NSString* identifier = [bundle bundleIdentifier];
identifier = [bundle bundlePath];
return identifier ? util::MakeString(identifier) : "(missing bundle identifier)";
```

Your version is much more explicit, though, and you can leave it as is.

Reply:
The second line of your suggestion doesn't work since we only want to do that if the `-bundleIdentifier` returned `nil`.

In general I'm not in favor of relying on nil dispatch. It's too easy to accidentally do when it's not intended and when reading code later it's hard to understand if it's intended or not. It also hides the conditional behavior in a way that's easy to overlook.

I did make the final line the ternary though.